### ISSUE TITLE
no-use-before-define should not err on export declarations

### ISSUE THREAD

**Issue by proteriax** [typescript-eslint/typescript-eslint#1738]
**What code were you trying to parse?**

```ts
// Please put code here
export { foo }
function foo() {}
```

**What did you expect to happen?**
No error.

**What actually happened?**
'foo' was used before it was defined. eslint(@typescript-eslint/no-use-before-define)

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `2.23.0` |
| `@typescript-eslint/parser`               | `2.23.0` |
| `TypeScript`                              | `3.8.3` |
| `ESLint`                                  | `6.8.0` |
| `node`                                    | `13.10.1` |

**Comment by bradzacher** [comment]
this is working as intended.
the rule does not ignore anything by default.
if you want it to ignore functions, then there is an option for that
https://eslint.org/docs/rules/no-use-before-define#options
